### DISCUSSION TITLE
make "fake cursor" still trigger hover events

### DISCUSSION THREAD

**Discussion by dragoncoder047** [kaplayjs/kaplay#330] [Q&A] [answered]
So I decided that my game would not force people to use their mouse. To do that I requested pointer lock (i.e. `setCursorLocked(true)`), added a game object with a custom cursor sprite and a z-index of infinity, and programmed it to move when the mouse moves, but also move according to the right gamepad stick.

Unfortunately with pointer lock turned on Kaplay no longer knows where the pointer is and so the area component can't fire its `onHover` event.

How do I tell Kaplay that my synthesized pointer should be used for the hover events and/or feed the synthesized pointer's position into the mouse-hover logic?

**Comment by lajbel** [comment]
Oh! Area doesn't fire up an event when the object is hovered. 

```js
        isHovering(this: GameObj) {
            const mpos = isFixed(this)
                ? k.mousePos()
                : k.toWorld(k.mousePos());
            return this.hasPoint(mpos);
        },
```

I would say you can mimic it triggering you the event (example: `obj.trigger("hover
)` when the mimic mouse collides with an area), but you can't, because there's no event.

So, the fix is either add events for hover or add in isHovering a check for an object with the tag "mimicMouse" or something like that

**Reply by dragoncoder047** [reply]
If you are going to make the hovering code check for game objects with certain tags as well, I would recommend that the tag it checks be configurable via the kaplay global options. That way if some developer is trying to add this functionality in later and the default tag conflicts with existing objects that aren’t meant to be mouse cursors, they can just change it once in the kaplay options rather than having to change it everywhere in the rest of their code and now the terminology is different and they have to change how they think.

**Reply by lajbel** [reply, marked as answer]
yeah, I was thinking in a component anyway. But probably not doing the trigger events thing because it would implicate check isHovering on an update frame, and if you are not using it, will cause frame drops easily/
